saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAttributeValueParser.java

```diff
@@ -105,7 +105,23 @@ public Object parse(XMLEventReader xmlEventReader) throws ParsingException {
             return StaxParserUtil.getElementText(xmlEventReader);
         }
 
-        throw logger.parserUnknownXSI(typeValue);
+        // KEYCLOAK-18417: Simply ignore unknown types
+        logger.warn("Skipping attribute value of unsupported type " + typeValue);
+        skipElementAndSubElements(xmlEventReader);
+        return "";
+    }
+
+    private void skipElementAndSubElements(final XMLEventReader xmlEventReader) throws ParsingException {
+        try {
+            while (xmlEventReader.peek() != null && !xmlEventReader.peek().isEndElement()) {
+                if (((XMLEvent) xmlEventReader.next()).isStartElement()) {
+                    // Recursively skip nested elements
+                    skipElementAndSubElements(xmlEventReader);
+                }
+            }
+        } catch (Exception e) {
+            throw logger.parserError(e);
+        }
```

Comment:
This is insufficient e.g. when the SAML document contains comments and such, and would lead to infinite loop.